### ISSUE TITLE
Gesture with options?

### ISSUE THREAD

**Issue by mastastealth** [chriswessels/meteor-hammer#12]
Awesome package, thanks a lot for this!

I've been wondering how to add a few options to my gestures. For example rather than a simple:

```"swiperight .element" : function() ...```

I'm wondering how I can attach the options to detecting a [2 finger (pointer?) swipe](http://hammerjs.github.io/recognizer-swipe/), velocity, etc. Is this currently possible?

**Comment by chriswessels** [comment]
Hey @mastastealth,

It depends on what you want to do. chriswessels:hammer creates a single instance of `Hammer.Manager` on the body element (which you can set options on). You could try something like this:

```javascript
var manager = $('body').data('hammer'),
    swipeRecogniser = manager.get('swipe');

swipeRecogniser.set({
  velocity: 0.8,
  pointers: 2
});
```

If you need to set options that shouldn't be application-wide, you'll need to create a separate instance of `Hammer.Manager`. chriswessels:hammer bundles in the Hammer.js jQuery plugin, so this is pretty easy: `template.$('#element').hammer({ /* Options */});`.

I'm open to any ideas for making this easier.

Hope this helps.

Chris

**Comment by mastastealth** [comment]
Hmm, I'm not sure if the separate is what I need or not, I understand the global edit but that's definitely not it either.

Basically I'd like to be able to do something along the lines of:
```
"swiperight .element" : function() { 
    // Do something
},
"2fingerswiperight .element" : function() { 
    // Do something else
}
```
One option is to add a few more default aliases for those gestures, but I think that might get messy and I'm not sure there's a simple way to prefix them ("2f-"?). Alternatively, how about a way that I can define my own gestures? So I could do something like

Another thing would maybe something like:
```
var myGestureEvent = Hammer.Swipe({
    pointers: 2, velocity: 0.5
});

"myGestureEvent .element" : function() { 
    // Do what I want, with my options
}
```

Basically, I'd like to define a 1 finger swipe along with a 2 finger swipe somehow. :D

**Comment by chriswessels** [comment]
Hey @mastastealth,

This is completely untested, but it looks like the options that are passed into the constructor for `Hammer.Swipe` and other Recognizers allows you to set the event name. Give this a try:

```javascript
// client/app.js
Meteor.startup(function () {
  var manager = $('body').data('hammer');
  var twoFingerSwipe = new Hammer.Swipe({
    event: '2fingerswipe',
    pointers: 2,
    velocity: 0.5
  });
  manager.add(twoFingerSwipe);
});
```

Then you should be able to

```javascript
"2fingerswiperight .element" : function() { 
    // Do something else
}
```

Let me know if this works... Might be worth adding to the docs if it does.

Chris

**Comment by mastastealth** [comment]
Indeed it does! Awesome, thanks! :+1:

**Comment by chriswessels** [comment]
Great stuff @mastastealth. I'll update the docs to make this kind of usage clearer.

Closing the issue...

Chris
